JASP-Engine/JASP/R/multinomialtest.R

```diff
@@ -15,546 +15,447 @@
+  #   chisqResults:
+  #   
+  # Return:
+  #   No return, stores jaspResults Object (main table results)
+  
+  results <- list()
+
+  # Compute Results
+  .chisquareTest(jaspResults, dataset, options)
+  # Get Results
+  chisqResults <- jaspResults[["stateChisqResults"]]$object
+  
+  if(!is.null(chisqResults)){
+    # fill in results one row at a time
+    for (r in 1:length(chisqResults)) {
+      df       <- chisqResults[[r]][["parameter"]][["df"]]
+      pVal     <- chisqResults[[r]][["p.value"]]
+      chisqVal <- chisqResults[[r]][["statistic"]][["X-squared"]]
+      # This happens when the simulate.p.value option is TRUE, i.e. using monte carlo
+      if (is.na(df)) 
+        df <- "-" 
+      results[[r]] <- list(case = names(chisqResults)[r],
+                           chisquare = chisqVal, df = df, p = pVal)
+      if (options$VovkSellkeMPR)
+        results[[r]][["VovkSellkeMPR"]] <- .VovkSellkeMPR(pVal)
+      jaspResults[["chisqTable"]]$addRows(results[[r]])
+    }
+  }
 }
 
+.multinomDescriptivesTableFill <- function(jaspResults, options, chisqResults){
+  # Turn chi-square object into jaspResults 
+  # object and creates results for descripties table
+  #
+  # Args:
+  #   jaspResults:
+  #   dataset: input dataset
+  #   options: user options
+  #   
+  # Return:
+  #   No return, stores jaspResults Object (descriptives table results)
+  
+  results <- list()
+  footnotes <- list()
+
+  observed <- chisqResults[[1]][["observed"]]
+  if (options$countProp == "descCounts")
+    observed <- as.integer(observed)
+  else 
+    observed <- as.numeric(observed)/sum(observed)
+  
+  nms <- names(chisqResults)
+  tableFrame <- data.frame(
+    factor   = names(chisqResults[[1]][["observed"]]),
+    observed = observed,
+    stringsAsFactors = FALSE
+  )
+  if (options$countProp == "descCounts")
+    for (r in chisqResults)
+      tableFrame <- cbind(tableFrame, round(r[["expected"]]))
+  else
+    for (r in chisqResults){
+      div <- sum(chisqResults[[1]][["observed"]])
+      tableFrame <- cbind(tableFrame, r[["expected"]]/div)
+    }
+  
+  if (length(nms) == 1)
+    colnames(tableFrame)[-(1:2)] <- "expected"
+  else 
+    colnames(tableFrame)[-(1:2)] <- nms
+  
+  # Add confidenceInterval to the tableFrame
+  if (options$confidenceInterval){
+    ciDf <- .multComputeCIs(chisqResults[[1]][["observed"]], 
+                            options$confidenceIntervalInterval, 
+                            scale = options$countProp)
+    tableFrame <- cbind(tableFrame, ciDf)
+    message <- "Confidence intervals are based on independent binomial distributions."
+    results[["footnotes"]][["CImessage"]] <- message
+    if (any(is.nan(unlist(tableFrame[, c('lowerCI', 'upperCI')])))){
+      message <- "Could not compute confidence intervals."
+      results[["footnotes"]][["ciComputeError"]] <- message
+    }
+  }
+  
+  for (i in 1:nrow(tableFrame))
+    jaspResults[["descriptivesTable"]]$addRows(as.list(tableFrame[i,]))
```

Comment:
```suggestion
    jaspResults[["descriptivesTable"]]$setData(tableFrame)
```
is a little cleaner.